pkg/controller/core/capacity_controller.go

```diff
@@ -47,47 +55,121 @@ func NewCapacityReconciler(cache *capacity.Cache) *CapacityReconciler {
-	// No-op. All work is done in the event handler.
+	var capObj kueue.Capacity
+	if err := r.client.Get(ctx, req.NamespacedName, &capObj); err != nil {
+		// we'll ignore not-found errors, since there is nothing to do.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	log := ctrl.LoggerFrom(ctx).WithValues("capacity", klog.KObj(&capObj))
+	ctx = ctrl.LoggerInto(ctx, log)
+	log.V(2).Info("Reconciling Capacity")
+
+	usage, workloads, err := r.cache.Usage(&capObj)
+	if err != nil {
+		log.Error(err, "Failed getting usage from capacity cache")
+		// This is likely because the capacity was recently removed and we didn't
+		// process that event yet.
+		return ctrl.Result{}, err
+	}
+	// Shallow copy enough for now.
+	oldStatus := capObj.Status
+	capObj.Status.UsedResources = usage
+	capObj.Status.RunningWorkloads = int32(workloads)
+	if !equality.Semantic.DeepEqual(oldStatus, capObj.Status) {
+		err = r.client.Status().Update(ctx, &capObj)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
 	return ctrl.Result{}, nil
 }
 
 func (r *CapacityReconciler) Create(e event.CreateEvent) bool {
-	c := e.Object.(*kueue.Capacity)
+	c, match := e.Object.(*kueue.Capacity)
+	if !match {
+		// No need to interact with the cache for other objects.
+		return true
+	}
 	log := r.log.WithValues("capacity", klog.KObj(c))
 	log.V(2).Info("Capacity create event")
 	ctx := ctrl.LoggerInto(context.Background(), log)
 	if err := r.cache.AddCapacity(ctx, c); err != nil {
 		log.Error(err, "Failed to add capacity to cache")
 	}
-	return false
+	return true
 }
 
 func (r *CapacityReconciler) Delete(e event.DeleteEvent) bool {
-	c := e.Object.(*kueue.Capacity)
+	c, match := e.Object.(*kueue.Capacity)
+	if !match {
+		// No need to interact with the cache for other objects.
+		return true
+	}
 	r.log.V(2).Info("Queue delete event", "capacity", klog.KObj(c))
 	r.cache.DeleteCapacity(c)
-	return false
+	return true
 }
 
 func (r *CapacityReconciler) Update(e event.UpdateEvent) bool {
-	c := e.ObjectNew.(*kueue.Capacity)
-	log := r.log.WithValues("queue", klog.KObj(c))
+	c, match := e.ObjectNew.(*kueue.Capacity)
+	if !match {
+		// No need to interact with the cache for other objects.
+		return true
+	}
+	log := r.log.WithValues("capacity", klog.KObj(c))
 	log.V(2).Info("Capacity update event")
 	if err := r.cache.UpdateCapacity(c); err != nil {
 		log.Error(err, "Failed to update capacity in cache")
 	}
-	return false
+	return true
 }
 
 func (r *CapacityReconciler) Generic(e event.GenericEvent) bool {
 	r.log.V(3).Info("Ignore generic event", "obj", klog.KObj(e.Object), "kind", e.Object.GetObjectKind().GroupVersionKind())
-	return false
+	return true
+}
+
+type assignedWorkloadHandler struct{}
+
+func (h *assignedWorkloadHandler) Create(e event.CreateEvent, q workqueue.RateLimitingInterface) {
+	w := e.Object.(*kueue.QueuedWorkload)
+	if w.Spec.AssignedCapacity != "" {
+		q.Add(requestForWorkloadCapacity(w))
```

Comment:
I know this is related to kubebuilder/controller-runtime, but it would be good to add a comment explaining that those adds trigger the execution of the Reconcile logic.

Reply:
Done